src/evaluate/evaluator/utils.py

```diff
@@ -16,3 +16,24 @@ def __getitem__(self, i):
 
     def __iter__(self):
         return (self.dataset[i][self.key] for i in range(len(self)))
+
+
+class DatasetColumnTextClassification(list):
+    """Helper class to avoid loading a dataset column into memory when accessing it."""
+
+    def __init__(self, dataset: Dataset, key: str, key2: str):
+        self.dataset = dataset
+        self.key = key
+        self.key2 = key2
+
+    def __len__(self):
+        return len(self.dataset)
+
+    def __getitem__(self, i):
+        return {"text": self.dataset[i][self.key], "text_pair": self.dataset[i][self.key2] if self.key2 else None}
```

Comment:
maybe we can make this more general such that `"text"` and `"text_pair"` can also be configured and are not hardcoded?